### ISSUE TITLE
Add tutorial demonstrating restart

### ISSUE THREAD

**Issue by aprilnovak** [neams-th-coe/cardinal#381]
Users frequently need to restart Cardinal simulations, which involves restarting NekRS (from a field file) as well as the coupled MOOSE application, from Exodus. We should make a tutorial to describe this process.

tag @ahuxford

**Comment by ahuxford** [comment]
I know how to do Checkpoint, N to N restarting from [MOOSE Restart and Recover](https://mooseframework.inl.gov/application_usage/restart_recover.html) for restarting the coupled MOOSE app. I can make the tutorial files for this.

Are there any other methods for restarting we need to have in the tutorial?

**Comment by aprilnovak** [comment]
No, I think that covers it - thank you for helping out with this! Let's also provide a link to that MOOSE documentation page so we can direct users to more information beyond what we provide in the tutorial.

Once you have your part, I'll add a section about restarting from the OpenMC side of things.

**Comment by aprilnovak** [comment]
Thank you for helping out with this!
